### ISSUE TITLE
Error contain form CashePovider

### ISSUE THREAD

**Issue by Sirgalas** [doctrine/cache#275]
Good afternoon
I use the library https://github.com/Roave/DoctrineSimpleCache
RedScache works with your library and with the permission of your class
$ Redistributed > connections ( 'local', 6379);
$ $ cache = new \ Doctrine \ Common \ Cache \ Redis Cache ();
$ $ Cache-> set Redis ($ Redis);
new \ BrowscapPHP \ Browscap (new SimpleCacheAdapter ($ cashe), new NullLogger ());
here. https://github.com/Roave/DoctrineSimpleCache/blob/2551f4cfc0aa6f6ad08e86398c193e12292104b1/src/SimpleCacheAdapter.php#L200 they use your method contains, which for some reason returns a number (0,1) from your own library https://github.com /doctrine/cache/blob/5e65fc393ecc17e395ac44d7b855dcf5e6864f34/lib/Doctrine/Common/Cache/Cache.php#L38 is bool

**Comment by asgrim** [comment]
Alright, so this does indeed look like a bug here.

https://github.com/phpredis/phpredis/#exists

Documentation here states the return values can be `0` or `1` for single keys, or any number for multiple key checks.

As `doctrine/cache` only checks a single key at once, it _should_ be safe to just cast to a `bool`, but if we want to be a bit more clever, we should check the number of keys passed (unless we are 100% sure it will always be just the one).

Basically, `doContains` should return a `bool` but it doesn't, it returns an `int`... and, further on the phpredis ext, it seems that older versions of the ext returned `bool` always, so that's technically a BC break, but it's an ext, so ¯\_(ツ)_/¯ 

So basically, indeed this used to work in doctrine/cache but with newer versions of the ext, they broke BC and now return an int.

**Comment by asgrim** [comment]
Reproduction (needs Redis extension and a local Redis server...):

```
<?php

require_once __DIR__ . '/vendor/autoload.php';

$redis=new \Redis();
$redis->connect('localhost',6379);
$cache=new \Doctrine\Common\Cache\RedisCache();
$cache->setRedis($redis);

var_dump($cache->contains('foo'));
```

*Actual output*

```
int(0)
```

*Expected output*

```
bool(false)
```

**Comment by Ocramius** [comment]
This seems to be already fixed in `master`, thanks to https://github.com/doctrine/cache/pull/264

I'm gonna do a release for `doctrine/cache` then.

**Comment by asgrim** [comment]
Good spot :bat: man

**Comment by Ocramius** [comment]
It's released :shipit: https://github.com/doctrine/cache/releases/tag/v1.8.0

**Comment by asgrim** [comment]
Nice! Thanks for the report @Sirgalas :)

**Comment by Sirgalas** [comment]
Thanks for the fixes deleting your fork
